src/containers/Books/Books.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import { useHistory } from "react-router-dom";
+import BookCard from '../../components/BookCard/BookCard';
+import PaginationButton from '../../components/PaginationButton/PaginationButton';
+import SearchPanel from '../../components/SearchPanel/SearchPanel';
+import Loader from '../../components/Loader/Loader';
+import { getBooks } from '../../API/fetch';
+
+function Books() {
+  const [data, setData] = useState([]);
+  const [resultsValue, setResultsValue] = useState(0);
+  const [bookTitle, setBookTitle] = useState('');
+  const [sort, setSort] = useState('');
+  const [category, setCategory] = useState('');
+  const [loaderVisibility, setLoaderVisibility] = useState(true);
+  const history = useHistory();
+
+  const onFinish = (bookTitle, sort = 'relevance', category = 'all') => {
+    if (bookTitle) {
+      setLoaderVisibility(false);
+      setBookTitle(bookTitle);
+      setSort(sort);
+      setCategory(category);
+
+      getBooks(bookTitle, sort, category)
+        .then(data => {
+            setLoaderVisibility(true);
+            if (data.totalItems) {
+                setData(data.items);
+                setResultsValue(data.totalItems);
+            } else {
+                setData([]);
+                setResultsValue(data.totalItems);
+            }
```

Comment:
nonsensical if block with duplicate lines (30 and 33)

28-34 lines should be replaced with
'setData(data.totalItems ? data.items : []);
setResultValue(data.totalItems)'

moreover i think it can be 'setData(data.items || [])' in first line

also i dont like 'resultValue' name, it does not explain the basic meaning of the variable
maybe resultCount or smth like that will be better?

Reply:
Ok, will figure it out